dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatHistory.cs

```diff
@@ -69,6 +70,27 @@ public ChatHistory(IEnumerable<ChatMessageContent> messages)
     public void AddMessage(AuthorRole authorRole, ChatMessageContentItemCollection contentItems, Encoding? encoding = null, IReadOnlyDictionary<string, object?>? metadata = null) =>
         this.Add(new ChatMessageContent(authorRole, contentItems, null, null, encoding, metadata));
 
+    /// <summary>
+    /// Adds <see cref="ChatMessageContent"/> with content items./>
+    /// </summary>
+    /// <param name="authorRole">Role of the message author</param>
+    /// <param name="items">The content items.</param>
+    [Experimental("SKEXP0001")]
+    public void AddMessage(AuthorRole authorRole, params KernelContent[] items)
```

Comment:
It could be slightly more convenient for callers to use a shorter syntax instead of the long one (commented out):
<img width="677" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/a23f52a6-2afe-4489-a24a-34df9e03e0db">

Reply:
> It could be slightly more convenient for callers

Until they need to do something this doesn't support, like add AuthorName, and then they fall off a cliff. The params also means you can call this with no args beyond author role, e.g.
```C#
chatHistory.AddMessage(AuthorRole.User);
```
which is a pit of failure.

I don't think we should be adding any more of these "optimize for a very specific use case" helpers onto ChatHistory.

Reply:
Removed.